### PR TITLE
Fix import from latest Lr. Fixes #11846

### LINKED ISSUE #11846: Export as pdf produces only white page
**Did you buy darktable from an application store ?**
No

Apparently, exporting as pdf doesn't work :thinking:
This produce a pdf file with the same number of pages as selected picture number but they are all white and contains no picture.
I don't know if it needs additional lib for that.
I have tried many settings and it's always the same.

**How to reproduce**
- In Lightable, select pictures and select PDF format in Export module
- Click export.
- Open the resulting PDF and see

Darktable version : current 3.9 
Os : Ubuntu 20.04

### PR DESCRIPTION
**Pull request by TurboGit**
if possible to have this in 2.4 it would be nice. with the windows version and the migration from lr to dt quite important I suppose this will help quite a lot of people.

### CODE REVIEW

File: src/develop/lightroom.c [hunk 1]
```diff
@@ -469,560 +1009,109 @@ void dt_lightroom_import(int imgid, dt_develop_t *dev, gboolean iauto)
 //     return;
 //   }
 
-  xmlXPathFreeObject(xpathObj);
-  xmlXPathFreeContext(xpathCtx);
+  // let's now parse the needed data
 
-  // Go safely to Description node
+  lr_data_t data;
```
Comment (houz) [inline src/develop/lightroom.c]: Why all the `memset` instead of a single `lr_data_t data = {0};`?

Also, I hope it's not too big for the stack. @LebedevRI might know.

File: src/develop/lightroom.c [hunk 2]
```diff
@@ -469,560 +1009,109 @@ void dt_lightroom_import(int imgid, dt_develop_t *dev, gboolean iauto)
...
-    {
-      xmlNodePtr tagNode = entryNode;
+  data.rating = 0;
+  data.has_rating = FALSE;
 
-      tagNode = tagNode->xmlChildrenNode;
-      tagNode = tagNode->next;
-      tagNode = tagNode->xmlChildrenNode;
-      tagNode = tagNode->next;
+  data.lat = 0;
```
Comment (houz) [inline src/develop/lightroom.c]: Those should be `NAN`.

File: src/develop/lightroom.c [hunk 3]
```diff
@@ -469,560 +1009,109 @@ void dt_lightroom_import(int imgid, dt_develop_t *dev, gboolean iauto)
...
+      snprintf(expr, sizeof(expr), "//%s:*", names[i]);
+      _handle_xpath(dev, doc, imgid, xpathCtx, (const xmlChar *)expr, &data);
 
-          if(sscanf((const char *)value, "%d, %d", &(curve_pts[n_pts][0]), &(curve_pts[n_pts][1]))) n_pts++;
-          xmlFree(value);
-        }
-        if(n_pts == MAX_PTS) break;
-        tcNode = tcNode->next;
-      }
+      /* Lr up to 6.0 (attributes) */
```
Comment (houz) [inline src/develop/lightroom.c]: Can a Lr Xmp contain both 6.0 attributes and newer ones? If yes, it might make sense to check the old ones first and let the new ones overwrite them afterwards.
If they are mutually exclusive then ignore this comment.

File: src/develop/lightroom.c (hunk 1 above)
Comment (LebedevRI) [reply src/develop/lightroom.c]: The simplest/dumbest code is *usually* the best one.
As far as i can tell there is no difference whatsoever between `{}` vs `memset()`.
https://godbolt.org/g/qMc46p
But the `memset()` is worse because that is much more lines, symbols; and you need to specify sizes, offsets.

TLDR: just do what @houz suggested `lr_data_t data = {0};`

File: src/develop/lightroom.c (hunk 3 above)
Comment (TurboGit) [reply src/develop/lightroom.c]: No, this won't happen. The new XMP I've seen are fully converted to new format.

File: src/develop/lightroom.c (hunk 2 above)
Comment (TurboGit) [reply src/develop/lightroom.c]: Ok, will do and test.

File: src/develop/lightroom.c (hunk 1 above)
Comment (TurboGit) [reply src/develop/lightroom.c]: Indeed, better. Fixing.

Comment (TurboGit) [reply src/develop/lightroom.c]: Also I don't think there is change on the stack usage, the variables were there before.

**Comment by garrett** [comment]
Wow! Thanks! This will help me and a few friends who are all switching away from LR.

(FWIW: It was long overdue for me; I kept around a proprietary OS just for LR; I use Linux and FOSS for everything else.)

Does this also import locations like city, country, etc.? When trying to bring over Lightroom data to Darktable via XMP, it seemed that some of my metadata, most notably the location, was ignored.

**Comment by garrett** [comment]
(I'm happy to contribute some of my XMP files somewhere if you'd like some examples.)

File: src/develop/lightroom.c [hunk 4]
```diff
@@ -469,560 +1009,83 @@ void dt_lightroom_import(int imgid, dt_develop_t *dev, gboolean iauto)
...
-      pcz.equalizer_y[0][0] = 0.5 + lfactor * (float)v / 200.0;
-    }
-    else if(!xmlStrcmp(attribute->name, (const xmlChar *)"LuminanceAdjustmentOrange"))
-    {
-      int v = atoi((char *)value);
-      if(v != 0) has_colorzones = TRUE;
-      pcz.equalizer_y[0][1] = 0.5 + lfactor * (float)v / 200.0;
+  // let's now parse the needed data
+
+  lr_data_t data = { 0 };
```
Comment (LebedevRI) [inline src/develop/lightroom.c]: Okay, yeah, except the warnings.
I think one `memset()` over the entire `struct` should be fine still.

**Comment by TurboGit** [comment]
@garrett no city or country import. this is just a fix for the new Lr 7.x format. No new feature at this stage.

File: src/develop/lightroom.c (hunk 4 above)
Comment (TurboGit) [reply src/develop/lightroom.c]: @LebedevRI houz said that a single "{ 0 }" was ok and on my side I have no warnings. Maybe a different compiler? I'm using GCC 7.2.

Comment (LebedevRI) [reply src/develop/lightroom.c]: https://travis-ci.org/darktable-org/darktable/builds/312355224?utm_source=github_status&utm_medium=notification

Comment (houz) [reply src/develop/lightroom.c]: Well, just to be safe we should use memset then.

Comment (TurboGit) [reply src/develop/lightroom.c]: Ok, otherwise using "{ { 0 } }" works fine. But I'll use one memset().

**Comment by TurboGit** [comment]
@garrett and moreover there is no city or country field in darktable, so I'm not sure how this would fit anyway.

**Comment by garrett** [comment]
Oh that's a shame. Locations are an important part of IPTC, which is integrated into EXIF and XMP, and I'm pretty sure most other photographers also like to tag where their photos were taken.

(Ah well, sorry for being a bit off-topic. Thanks again for your contributions!)

**Comment by TurboGit** [comment]
I don't follow you. dt does import GPS location. But you were talking about city and country. There is no such field in dt, yet with the GPS data the location is quite precise. So now I'm not sure to understand your need.
